syntax = "proto3";
package aumos.events.v1;

option python_package = "aumos_proto.events.v1";

import "google/protobuf/timestamp.proto";
import "google/protobuf/struct.proto";

// Standard envelope for all agent-to-agent messages
message AgentEnvelope {
  string message_id = 1;
  string tenant_id = 2;
  string source_agent_id = 3;
  string target_agent_id = 4;
  string conversation_id = 5;
  string message_type = 6;  // task_request, task_response, observation, action
  google.protobuf.Struct payload = 7;
  int32 privilege_level = 8;  // 1-5
  google.protobuf.Timestamp created_at = 9;
  string correlation_id = 10;
  map<string, string> metadata = 11;
  int32 ttl_seconds = 12;
  int32 retry_count = 13;
}
